### ISSUE TITLE
False negative with `Lint/SymbolConversion` in combination with string interpolation

### ISSUE THREAD

**Issue by Earlopain** [rubocop/rubocop#12373]
With code like this I expect `Lint/SymbolConversion` to trigger an offence. Interpolation isn't currently considered.

```rb
foo = "bar"
"#{foo}_baz".to_sym
=> :bar_baz
```

Autocorrect would be the following which is equivalent:

```rb
:"#{foo}_baz"
```

## RuboCop version

```
1.55.1 (using Parser 3.2.2.3, rubocop-ast 1.29.0, running on ruby 3.2.2) [x86_64-linux-musl]
  - rubocop-erb 0.3.0
  - rubocop-rails 2.20.2
```

**Comment by shiroemons** [comment]
以下の2つのlintに引っかかっている
- [Style/RedundantParentheses](https://www.rubydoc.info/github/bbatsov/RuboCop/RuboCop/Cop/Style/RedundantParentheses)
  - https://github.com/rubocop/rubocop/issues/12311
- [Lint/SymbolConversion](https://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Lint/SymbolConversion)
  - https://github.com/rubocop/rubocop/issues/12373
